### PR TITLE
Add syslog facility configuration

### PR DESCRIPTION
**Pull request by korayvt**
With this PR, syslog facility configuration is added.

The user can filter syslog messages by setting the facility.

### CODE REVIEW

**Comment by misi** [comment]
Many Thanks..

**Comment by Neustradamus** [comment]
@misi: Happy New Year 2022!

Can you look the @korayvt PR?

Thanks in advance.

**Comment by korayvt** [comment]
It seems that @misi  is busy with other things.

Anyone else able to merge the commits and release a new version?

File: src/apps/common/ns_turn_utils.c [hunk 1]
```diff
@@ -147,6 +147,48 @@ int turn_mutex_destroy(turn_mutex* mutex) {
...
+		    LOG_KERN , LOG_LOCAL0 , LOG_LOCAL1 ,
+		    LOG_LOCAL2 , LOG_LOCAL3 , LOG_LOCAL4 , LOG_LOCAL5 ,
+		    LOG_LOCAL6 , LOG_LOCAL7 , LOG_LPR , LOG_MAIL ,
+		    LOG_NEWS , LOG_USER , LOG_UUCP,
+		    LOG_AUTHPRIV,LOG_FTP,LOG_SYSLOG,
+		    0};
+
+static int syslog_facility = 0;
+
+static int str2facility(char *s)
```
Comment (ggarber) [inline src/apps/common/ns_turn_utils.c:170]: can you rename to `str_to_syslog_facility` (for consistency with other functions like `addr_to_string`)?

**Review by ggarber — APPROVED** [review]
Looks great, thank you!  Can you take a look at the tiny nit and i'll merge it?

Also if you can add to the description of the PR how you tested the change (if you tested not passing the value, passing a specific value and passing an invalid string for example) it would be nice.

Thank you!

File: src/apps/common/ns_turn_utils.c (hunk 1 above)
Comment (korayvt) [reply src/apps/common/ns_turn_utils.c]: I realized some incomplete coding in my previous commit and fixed them.
Thanks for the comment.

**Review by ggarber — APPROVED** [review]

**Comment by steffen-moser** [comment]
Unfortunately, this PR breaks the possibility to compile coturn in Oracle Solaris 11.4 (SRU 48). It is caused by the fact that LOG_FTP is not defined in Solaris' "/usr/include/sys/syslog.h". They define the following syslog facility codes:

```
#define LOG_KERN        (0<<3)  /* kernel messages */
#define LOG_USER        (1<<3)  /* random user-level messages */
#define LOG_MAIL        (2<<3)  /* mail system */
#define LOG_DAEMON      (3<<3)  /* system daemons */
#define LOG_AUTH        (4<<3)  /* security/authorization messages */
#define LOG_SYSLOG      (5<<3)  /* messages generated internally by syslogd */
#define LOG_LPR         (6<<3)  /* line printer subsystem */
#define LOG_NEWS        (7<<3)  /* netnews subsystem */
#define LOG_UUCP        (8<<3)  /* uucp subsystem */
#define LOG_AUTHPRIV    (10<<3) /* sensitive/private security/auth messages */
#define LOG_AUDIT       (13<<3) /* audit subsystem */
#define LOG_CRON        (15<<3) /* cron/at subsystem */
        /* other codes through 15 reserved for system use */
#define LOG_LOCAL0      (16<<3) /* reserved for local use */
#define LOG_LOCAL1      (17<<3) /* reserved for local use */
#define LOG_LOCAL2      (18<<3) /* reserved for local use */
#define LOG_LOCAL3      (19<<3) /* reserved for local use */
#define LOG_LOCAL4      (20<<3) /* reserved for local use */
#define LOG_LOCAL5      (21<<3) /* reserved for local use */
#define LOG_LOCAL6      (22<<3) /* reserved for local use */
#define LOG_LOCAL7      (23<<3) /* reserved for local use */
```

So omitting LOG_FTP should fix this. I really doubt that it is needed.

**Pull request merged**